system/lib/wasmfs/file.h

```diff
@@ -242,6 +243,8 @@ class Directory : public File {
   //  3. Implementing `getName`, since it cannot be implemented in terms of the
   //     dcache.
   virtual bool maintainsFileIdentity() { return false; }
+  
+  friend class IgnoreCaseDirectory;
```

Comment:
Yikes! This looks like a bad abstraction leak. Why do we need this?

Reply:
I know. We need to forward some calls to `baseDirectory` in `IgnoreCaseDirectory`. For example, `insertDirectory()`:
https://github.com/emscripten-core/emscripten/blob/7d5b438340dea4814778b9b34399053f559b0524/system/lib/wasmfs/backends/ignore_case_backend.cpp#L53
We could have used `Directory::Hanlde` in that case. But it seems bad because:
* Working with `Directory::Handle` means some additional code would run instead of directly calling `Directory` method. 
  * `insertDirectory()` for a root won't work because the directory doesn't have parent: https://github.com/emscripten-core/emscripten/blob/7d5b438340dea4814778b9b34399053f559b0524/system/lib/wasmfs/file.cpp#L115
  * `dcache` of a child will be used but we don't need it.
* `baseDirectory` is a part of the `IgnoreCaseDirectory` object so locking it seems redundant

Do you have any better ideas how to solve that without a need of a friend class?

Reply:
For now I would just use `Directory::Handle` like normal and set the parent manually so that `insertDirectory` works. It would be good to add TODO comments about making that nicer, though. These are good usability problems to surface and fix before we create a bunch more virtual backends.